Automattic/jetpack
# Donations block: fix race condition when updating currency

Fixes #22838
When we change the currency of a donations block we can save the post before we've created the plan for the given currency, this causes the block to display the currency set by the user but we don't have a subscription plan for it so the previous subscription plan is shown when paying through the payment form.

#### Changes proposed in this Pull Request:

* Now the update button of the post is blocked until we complete the operation of creating the subscription plan for the selected currency.

#### Does this pull request change what data or activity we track or use?
No

#### Other information:


#### Testing instructions:






* Add a donations block to a new post and publish the post.
* Change currency to another currency, e.g. Australian dollar
* Update the post as fast as you can (press the update button when available)
* View the page
* Choose an amount
* Press Donate
* The currency should match the currency of the block.

#### Test evidence:
https://user-images.githubusercontent.com/1989914/188640556-157e2667-25d9-476c-b56f-7e53dbe4c62c.mp4
